test/e2e_tests/specs/Block/block.spec.ts

```diff
@@ -0,0 +1,344 @@
+
+const test = baseTest.extend<{userBPageManager: PageManager; connectUsersViaAPI: boolean}>({
+  userBPageManager: async ({browser}, use) => {
+    const context = await browser.newContext();
+    const page = await context.newPage();
+    const manager = PageManager.from(page);
+    await use(manager);
+    await context.close();
+  },
+  connectUsersViaAPI: [true, {option: true}],
+});
+
+test.describe('User Blocking', () => {
+  test.describe('Block: User A and User B are NOT in the same team', () => {
+    const setupTeamForUser = async (api: ApiManagerE2E, userToInvite: User, teamSuffix: string) => {
+      let teamOwner = getUser();
+
+      const createdOwner = await api.createTeamOwner(teamOwner, `Block Test Team ${teamSuffix}`);
+      teamOwner = {...teamOwner, ...createdOwner};
+
+      const invitationId = await api.team.inviteUserToTeam(userToInvite.email, teamOwner);
+      const invitationCode = await api.brig.getTeamInvitationCodeForEmail(teamOwner.teamId, invitationId);
+
+      await api.createPersonalUser(userToInvite, invitationCode);
+    };
+
+    test.beforeEach(
+      'User A and User B are in different teams',
+      async ({api, pageManager: userAPageManager, userBPageManager, connectUsersViaAPI}) => {
+        userA = getUser();
+        userB = getUser();
+
+        await test.step('Preconditions: Creating separate teams for User A and User B', async () => {
+          await Promise.all([setupTeamForUser(api, userA, 'A'), setupTeamForUser(api, userB, 'B')]);
+        });
+
+        if (connectUsersViaAPI) {
+          await api.connectUsers(userA, userB);
+        }
+
+        await userAPageManager.openMainPage();
+        await userBPageManager.openMainPage();
+
+        await test.step('Preconditions: Signing in User A and User B', async () => {
+          await Promise.all([loginAndSetup(userA, userAPageManager), loginAndSetup(userB, userBPageManager)]);
+        });
+      },
+    );
+
+    test.afterEach(async ({api}) => {
+      await tearDownAll(api);
+    });
+
+    test(
+      'I want to cancel blocking a 1on1 conversation from conversation list 0',
+      {tag: ['@TC-137', '@regression']},
+      async ({pageManager: userAPageManager}) => {
+        const {pages: userAPages, modals: userAModals} = userAPageManager.webapp;
+
+        // Step 1: User A opens conversation with User B
+        await userAPages.conversationList().openConversation(userB.fullName);
+        // Step 2: User A opens the options menu for user B
+        await userAPages.conversationList().clickConversationOptions(userB.fullName);
+        // Step 3: User A opens modal and clicks 'Block' button
+        await userAPages.conversationList().clickBlockConversation();
+        // Step 4: User A clicks 'Cancel' button
+        await userAModals.blockWarning().clickCancel();
+        // Step 5: Conversation is still present, and User A can open it
+        await userAPages.conversationList().openConversation(userB.fullName);
```

Comment:
This test doesn't have any assertion, you just rely on the call to openConversation to not fail but this is not a valid way of testing the behavior. For example: If you change line 104 to call `clickBlock()` instead of `clickCancel()` the test still passes.

Tip: Always try to make your test fail once before making it pass ;)